### PR TITLE
[ticket/13694] Allow modifying the Postgres native author search query for results

### PR DESCRIPTION
**Pull request by brunoais**
- [ ] based on https://github.com/phpbb/phpbb/pull/3479

https://tracker.phpbb.com/browse/PHPBB3-13694

### CODE REVIEW

File: phpBB/phpbb/search/fulltext_postgres.php [hunk 1]
```diff
@@ -595,6 +603,55 @@ public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $
 
 		$m_approve_fid_sql = ' AND ' . $post_visibility;
 
+		/**
+		* Allow changing the query used to search for posts by author in fulltext_postgres
+		*
+		* @event core.search_native_author_count_query_before
```
Comment (Nicofuma) [inline phpBB/phpbb/search/fulltext_postgres.php]: postgres

File: phpBB/phpbb/search/fulltext_postgres.php [hunk 2]
```diff
@@ -595,6 +603,55 @@ public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $
...
+			'sql_sort',
+			'sort_days',
+			'sql_time',
+			'firstpost_only',
+			'ex_fid_ary',
+			'sql_fora',
+			'm_approve_fid_sql',
+			'start',
+		);
+		extract($this->phpbb_dispatcher->trigger_event('core.search_native_author_count_query_before', compact($vars)));
```
Comment (Nicofuma) [inline phpBB/phpbb/search/fulltext_postgres.php]: postgres

File: phpBB/phpbb/search/fulltext_postgres.php [hunk 3]
```diff
@@ -595,6 +603,55 @@ public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $
...
+		* @var	string	sort_dir			"a" for ASC or "d" dor DESC for the sort order used
+		* @var	string	sql_sort			The result SQL when processing sort_by_sql + sort_key + sort_dir
+		* @var	string	sort_days			Time, in days, that the oldest post showing can have
+		* @var	string	sql_time			The SQL to search on the time specifyed by sort_days
+		* @var	bool	firstpost_only		Wether or not to search only on the first post of the topics
+		* @var	array	ex_fid_ary			Forum ids that must not be searched on
+		* @var	array	sql_fora			SQL query for ex_fid_ary
+		* @var	string	m_approve_fid_sql	WHERE clause condition on post_visibility restrictions
+		* @var	int		start				How many posts to skip in the search results (used for pagination)
+		* @since 3.1.4-RC1
```
Comment (marc1706) [inline phpBB/phpbb/search/fulltext_postgres.php]: 3.1.5-RC1

**Comment by nickvergessen** [comment]
based on https://github.com/phpbb/phpbb/pull/3479

**Pull request merged**